experiment/autobumper/bumper/bumper.go

```diff
@@ -38,26 +42,197 @@ const (
+		return fmt.Errorf("--github-token is mandatory when --skip-pull-request is false")
+	}
+	if !o.SkipPullRequest && (o.GitHubOrg == "" || o.GitHubRepo == "") {
+		return fmt.Errorf("--github-org and --github-repo are mandatory when --skip-pull-request is false")
+	}
+	if (o.GitEmail == "") != (o.GitName == "") {
+		return fmt.Errorf("--git-name and --git-email must be specified together")
+	}
+
+	if o.TargetVersion != latestVersion && o.TargetVersion != upstreamVersion &&
+		o.TargetVersion != upstreamStagingVersion && !tagRegexp.MatchString(o.TargetVersion) {
+		return fmt.Errorf("invalid --target-version, must be one of %v",
+			[]string{latestVersion, upstreamVersion, upstreamStagingVersion, tagVersion})
+	}
+	if !o.BumpProwImages && !o.BumpTestImages {
+		return fmt.Errorf("at least one of --bump-prow-images and --bump-test-images must be specified")
+	}
+	if o.BumpProwImages && o.BumpTestImages && o.TargetVersion != latestVersion {
+		return fmt.Errorf("--target-version must be latest if you want to bump both prow and test images")
+	}
+	if o.BumpTestImages && (o.TargetVersion == upstreamVersion || o.TargetVersion == upstreamStagingVersion) {
+		return fmt.Errorf("%q and %q versions can only be specified to bump prow images", upstreamVersion, upstreamStagingVersion)
+	}
+
+	if len(o.IncludedConfigPaths) == 0 {
+		return fmt.Errorf("--include-config-paths is mandatory")
+	}
+
+	return nil
+}
+
+func Run(o *Options) error {
+	if err := validateOptions(o); err != nil {
+		return fmt.Errorf("error validating options: %v", err)
+	}
+
+	if err := cdToRootDir(); err != nil {
+		return fmt.Errorf("failed to change to root dir: %v", err)
+	}
+
+	images, err := updateReferences(
+		o.BumpProwImages, o.BumpTestImages, o.TargetVersion,
+		o.IncludedConfigPaths, o.ExcludedConfigPaths, o.ExtraFiles)
+	if err != nil {
+		return fmt.Errorf("failed to update image references: %v", err)
+	}
+
+	changed, err := hasChanges()
+	if err != nil {
+		return fmt.Errorf("error occurred when checking changes: %v", err)
+	}
+
+	if !changed {
+		logrus.Info("no images updated, exiting ...")
+		return nil
+	}
+
+	if o.SkipPullRequest {
+		logrus.Debugf("--skip-pull-request is set to true, won't create a pull request.")
+	} else {
+		sa := &secret.Agent{}
+		if err := sa.Start([]string{o.GitHubToken}); err != nil {
+			return fmt.Errorf("failed to start secrets agent: %v", err)
+		}
+
+		gc := github.NewClient(sa.GetTokenGenerator(o.GitHubToken), sa.Censor, github.DefaultGraphQLEndpoint, github.DefaultAPIEndpoint)
+
+		if o.GitHubLogin == "" || o.GitName == "" || o.GitEmail == "" {
+			user, err := gc.BotUser()
+			if err != nil {
+				return fmt.Errorf("failed to get the user data for the provided GH token: %v", err)
+			}
+			if o.GitHubLogin == "" {
+				o.GitHubLogin = user.Login
+			}
+			if o.GitName == "" {
+				o.GitName = user.Name
+			}
+			if o.GitEmail == "" {
+				o.GitEmail = user.Email
+			}
+		}
+
+		remoteBranch := "autobump"
```

Comment:
It's possible that there are multiple bumping PRs going on, i.e. oss-test-infra, can we allow more customized name?

Reply:
Good point. I have added one extra flag to allow users configure it, and the default value is `autobump`.